src/parser/jobs/mnk/modules/TwinSnakes.tsx

```diff
@@ -74,23 +77,23 @@ export default class TwinSnakes extends Module {
 	}
 
 	private onCast(event: CastEvent): void {
-		const action = getDataBy(ACTIONS, 'id', event.ability.guid) as TODO
+		const action = this.data.getAction(event.ability.guid)
 
 		// Only include GCDs
-		if (!action || !action.onGcd) {
+		if (!action?.onGcd) {
 			return
 		}
 
 		switch (action.id) {
 		// Ignore TS itself, plus Form Shift and maybe 6SS?
 		// We use gcdsSinceTS because we don't want to double count FPF
 		case (ACTIONS.TWIN_SNAKES.id):
-			if (this.twinSnake && !this.twinSnake.end && this.gcdsSinceTS < TWIN_SNAKES_CYCLE_LENGTH) {
+			if (!this.twinSnake?.end && this.gcdsSinceTS < TWIN_SNAKES_CYCLE_LENGTH) {
 				this.earlySnakes++
```

Comment:
This may not be a straight refactor here... https://xivanalysis.com/analyse/BM9hJdHwpxfFN6qg/11/6/ is showing an extra early refresh with this change in place, which looks to be a detection on the very first Twin Snake use.

Before the change, the if is going to return FALSE if this.twinSnake is null/undefined (from the first element), but now it will return TRUE if this.twinSnake is null/undefined and this.gcdsSinceTS is < CYCLE_LENGTH, since this.twinSnake?.end -> null if twinSnake is null/undefined, and !null -> true from a boolean logic perspective.

Reply:
Hm I'm just gonna revert this one then.